src/Entity/WebHook/WebHook.php

```diff
@@ -27,76 +27,83 @@ class WebHook
      *
      * @ORM\Column(length=64)
      *
-     * @JMS\Groups({"api"})
+     * @Assert\NotBlank
      */
     private $event;
 
     /**
-     * @var Collection|Callback[]
+     * This property is needed to manage to have the unique constraint "web_hook_event_client_id_unique" with Doctrine ORM.
      *
-     * @ORM\OneToMany(targetEntity="AppBundle\Entity\WebHook\Callback", mappedBy="webHook", cascade={"all"}, orphanRemoval=true)
+     * @var Client
      *
-     * @Assert\Valid
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\OAuth\Client", inversedBy="webHooks")
+     * @ORM\JoinColumn(nullable=false)
+     *
+     * @Assert\NotNull
+     */
+    private $client;
+
+    /**
+     * @var string[]
+     *
+     * @ORM\Column(type="json")
+     *
+     * @Assert\Count(min=1, minMessage = "Veuillez spécifier au moins une url de callback.")
      */
     private $callbacks;
 
-    public function __construct(Event $event, $callbacks = [])
+    public function __construct(Client $client = null, Event $event = null, $callbacks = [])
```

Comment:
Why all arguments are optional?

Reply:
Because we are coupled to sonata and forms. So when creating a new web hook sonata needs a clean web hook instance. I know I can override the object creation in the SonataAdmin class but it would just enforce the `$client` property.